src/main/scala/beam/agentsim/infrastructure/ScaleUpCharging.scala

```diff
@@ -0,0 +1,349 @@
+import beam.agentsim.events.RefuelSessionEvent.NotApplicable
+import beam.agentsim.events.SpaceTime
+import beam.agentsim.infrastructure.ChargingNetwork.ChargingVehicle
+import beam.agentsim.infrastructure.ChargingNetworkManager._
+import beam.agentsim.infrastructure.ParkingInquiry.{activityTypeStringToEnum, ParkingActivityType}
+import beam.agentsim.infrastructure.taz.{TAZ, TAZTreeMap}
+import beam.agentsim.scheduler.BeamAgentScheduler.{CompletionNotice, ScheduleTrigger}
+import beam.agentsim.scheduler.Trigger
+import beam.agentsim.scheduler.Trigger.TriggerWithId
+import beam.sim.config.BeamConfig.Beam.Agentsim
+import beam.utils.MathUtils.roundUniformly
+import beam.utils.{MathUtils, VehicleIdGenerator}
+import org.apache.commons.math3.distribution.EnumeratedDistribution
+import org.apache.commons.math3.random.MersenneTwister
+import org.apache.commons.math3.util.{Pair => CPair}
+import org.matsim.api.core.v01.Id
+import org.matsim.api.core.v01.population.Person
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.collection.JavaConverters._
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable
+import scala.language.postfixOps
+import scala.util.Random
+
+trait ScaleUpCharging extends {
+  this: ChargingNetworkManager =>
+  import ScaleUpCharging._
+
+  private lazy val rand: Random = new Random(beamConfig.matsim.modules.global.randomSeed)
+  private lazy val mersenne: MersenneTwister = new MersenneTwister(beamConfig.matsim.modules.global.randomSeed)
+  private lazy val cnmConfig: Agentsim.ChargingNetworkManager = beamConfig.beam.agentsim.chargingNetworkManager
+  private lazy val timeStepByHour = beamConfig.beam.agentsim.chargingNetworkManager.timeStepInSeconds / 3600.0
+  private lazy val virtualParkingInquiries: TrieMap[Int, ParkingInquiry] = TrieMap()
+  private lazy val vehicleRequests = mutable.HashMap.empty[(Id[TAZ], ParkingActivityType), List[VehicleRequestInfo]]
+
+  private lazy val scaleUpFactors: Map[ParkingActivityType, Double] = {
+    if (!cnmConfig.scaleUp.enabled) Map()
+    else {
+      Map(
+        ParkingActivityType.Home     -> cnmConfig.scaleUp.expansionFactor_home_activity,
+        ParkingActivityType.Init     -> cnmConfig.scaleUp.expansionFactor_init_activity,
+        ParkingActivityType.Work     -> cnmConfig.scaleUp.expansionFactor_work_activity,
+        ParkingActivityType.Charge   -> cnmConfig.scaleUp.expansionFactor_charge_activity,
+        ParkingActivityType.Wherever -> cnmConfig.scaleUp.expansionFactor_wherever_activity
+      )
+    }
+  }
+
+  private lazy val defaultScaleUpFactor: Double =
+    if (!cnmConfig.scaleUp.enabled) 1.0 else cnmConfig.scaleUp.expansionFactor_wherever_activity
+
+  override def loggedReceive: Receive = {
+    case t @ TriggerWithId(PlanParkingInquiryTrigger(_, requestId), triggerId) =>
+      log.debug(s"Received parking response: $t")
+      self ! virtualParkingInquiries.getOrElse(
+        requestId,
+        log.error(
```

Comment:
This will send a Unit back to self which will go to DeadLetters. Since this is logged here anyway then I'm ok with it - more of a note in case you wanted to adjust in any way.

Reply:
Fixed!